chapters/data-processing.tex

```diff
@@ -0,0 +1,725 @@
+and the \textbf{unit of observation}\sidenote{\textbf{Unit of observation:}
+	the unit described by the data. In datasets, it is ideally what each row represents. \url{
+		https://dimewiki.worldbank.org/wiki/Unit_of_Observation}}
+ may be ambiguous in many raw datasets.
+In this session, we will discuss the ideal format to handle data,
+which we call a \textit{tidy} format.
+
+
+\subsection{Unique identifiers}
+
+% Uniquely and fully identifying variable
+The first step in tidying data is to understand its unit of observation
+(what makes a row),
+and find out which variable or set of variables is the \textbf{unique identifier}\sidenote{\url{
+		https://dimewiki.worldbank.org/ID\_Variable\_Properties}} for each observation in your dataset.
+Ensuring that observations are uniquely and fully identified
+is arguably the most important step in data cleaning.
+It may be the case that the variables expected to be unique identifiers are either incomplete or contain duplicates.
+It is also possible for a dataset to have no unique identifier, 
+or that the identifier involves a long string that is difficult to work with, such as a full name.
+In such cases, cleaning begins by carefully creating a variable that uniquely identifies the data.
+Note that while modern survey tools create unique identifiers for each data submission,
+that is not the same as having a unique ID variable for each observation in the sample,
+as there can be multiple submissions for the same observation.
+The unique identifier will be used to link data on observations in one dataset 
+with data on the same observations in other datasets, 
+such as other rounds of data collection and different data sources.
+Therefore, you must keep a canonical dataset of all observations in your data.
+This \textbf{master dataset}\sidenote{\url{
+		https://dimewiki.worldbank.org/Master\_Data\_Set}}
+is discussed in detail in Chapter 3, 
+and contains a record of all units ever encountered for each relevant level of observation,
+as well as their corresponding unique identifiers.
+
+
+% Merchand
+Checking for duplicated entries and making sure that each observation has a unique identifier are usually part of data quality monitoring.
+The lack of a unique identifier should be addressed as soon as data is received,
+so you can keep track of observations.
+\texttt{ieduplicates} and \texttt{iecompdup},
+two Stata commands included in the \texttt{iefieldkit}
+package\index{iefieldkit},\sidenote{\url{https://dimewiki.worldbank.org/iefieldkit}}
+create an automated workflow to identify, correct and document
+occurrences of duplicate entries.
+One advantage of using \texttt{ieduplicates}\sidenote{\url{
+		https://dimewiki.worldbank.org/wiki/ieduplicates}} 
+to correct duplicated entries is that it creates \textit{duplicates reports}
+that record which corrections made and why.
+Even if you are not using this command,
+it is important to include similar records in the \texttt{Documentation} folder.
```

Comment:
do we need to make a specific recommendation to the Documentation folder? apart from the section on `iefolder` this book does not specify this specifically where something is saved.